--- mongodb-src-r2.6.12/SConstruct~	2016-03-21 18:09:46.000000000 +0200
+++ mongodb-src-r2.6.12/SConstruct	2017-04-04 11:01:42.056852157 +0300
@@ -879,7 +879,8 @@
         env.Append( LINKFLAGS=" -fprofile-arcs -ftest-coverage " )
 
     if optBuild:
-        env.Append( CCFLAGS=["-O3"] )
+        env.Append( CCFLAGS=ARGUMENTS.get("CXXFLAGS", "-O3").split() )
+        env.Append( CPPDEFINES=ARGUMENTS.get("CPPFLAGS", "").split() )
     else:
         env.Append( CCFLAGS=["-O0"] )
 
