--- mongodb-src-r4.0.5/SConstruct.orig	2018-12-31 09:58:59.358465265 +0100
+++ mongodb-src-r4.0.5/SConstruct	2018-12-31 10:36:23.448439637 +0100
@@ -1820,12 +1820,6 @@
         env.Append( CCFLAGS=["-fprofile-arcs", "-ftest-coverage"] )
         env.Append( LINKFLAGS=["-fprofile-arcs", "-ftest-coverage"] )
 
-    if optBuild and not optBuildForSize:
-        env.Append( CCFLAGS=["-O2"] )
-    elif optBuild and optBuildForSize: 
-        env.Append( CCFLAGS=["-Os"] )
-    else:
-        env.Append( CCFLAGS=["-O0"] )
 
     # Promote linker warnings into errors. We can't yet do this on OS X because its linker considers
     # noall_load obsolete and warns about it.
