--- mongodb-src-r2.0.6/SConstruct.orig	2012-06-13 13:02:51.000000000 +0200
+++ mongodb-src-r2.0.6/SConstruct	2012-06-13 13:03:02.000000000 +0200
@@ -196,7 +196,7 @@
 for shortName in getThirdPartyShortNames():
     add_option( "use-system-" + shortName , "use system version of library " + shortName , 0 , True )
 
-add_option( "use-system-all" , "use all system libraries " + shortName , 0 , True )
+add_option( "use-system-all" , "use all system libraries ", 0 , True )
 
 # --- environment setup ---
 
